lib/src/data/product.dart

```diff
@@ -1,29 +1,39 @@
+import "category.dart";
 import "types.dart";
 
 /// A product being sold on the Marketplace.
-/// 
+///
 /// A product can have multiple physical copies available for sale. When it is sold, it is not
 /// removed from the marketplace. Instead, the [quantity] is decremented (if it reaches zero, it is
 /// marked as "unavailable"). This way, customers can see past products and request the seller make
 /// more. If the seller wants to be done with a product, they can de-list it by setting [delisted].
 class Product {
   /// This product's unique Product ID.
   final ProductID id;
+
   /// The seller's unique Seller ID.
   final SellerID sellerID;
 
   /// The title or name of the product.
   final String title;
+
   /// The product's description.
   final String description;
+
   /// The price of this product
   final double price;
+
+  /// The category in which the Product belongs to
+  final Category category;
```

Comment:
I think you're correct with this. There can be a generalised category like "Under $10" and any item under $10 irrespective of whether it's an apparel or something else. I'll make changes to my local repo and change the implementation everywhere I've used Cateogory from Product class. I'm keeping this thread open so that when I push again, I'll update it over here and resolve this comment.